src/Elcodi/Admin/ProductBundle/Form/Type/CategoryType.php

```diff
@@ -82,51 +93,60 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $currentCategoryId = $builder->getData()->getId();
+        $parentEntityWhereClause = $currentCategoryId
+            ? "c.id <> $currentCategoryId AND c.root = 1"
```

Comment:
For escaping and doctrine cache purposes, better to do this in the query builder, like:
```php
$qb = $this
    ->categoryRepository
    ->createQueryBuilder('c')
    ->where('c.root = 1');

if ($currentCategoryId) {
    $qb
        ->andWhere('c.id <> :parent_category')
        ->setParameter('parent_category', $currentCategoryId);
}
```